### PR TITLE
Add Kitchen Tests

### PR DESCRIPTION
**Pull request by kierenevans**
Unfortunately due to public/private repo... things... github no longer detected my previous fork of this repo as a fork.

Re-raising https://github.com/inviqa/chef-magento-ng/pull/7 with 2 additional commits:
1. Remove config-driven-helper's git reference from Berksfile now that it is published on the supermarket
2. Create the `#{config_path}/app/etc/` directory if it doesn't exist to allow a successful converge of the etc-local recipe when used alone.

### CODE REVIEW

File: recipes/etc-local.rb [hunk 1]
```diff
@@ -30,6 +30,11 @@
       config_path = site['docroot']
     end
 
+    directory "#{config_path}/app/etc" do
+      recursive true
+      action :create
+    end
+
```
Comment (andytson) [inline recipes/etc-local.rb]: btw, I just realised I don't like this any more, it'll change the permissions of the folder to root, when capistrano recipe previously changed it to deploy

Comment (andytson) [inline recipes/etc-local.rb]: `action :create_if_not_exists` may be a compromise

Comment (andytson) [inline recipes/etc-local.rb]: doh, doesn't look like chef made that possible, I guess the alternative is a `not_if`

Comment (kierenevans) [inline recipes/etc-local.rb]: `If this value is not specified, existing owners remain unchanged and new owner assignments use the current user (when necessary).` from https://docs.chef.io/resource_directory.html

There is also the resource merging that occurs if the directory resource is defined twice, where the correct owner and group of the directory would be merged in.

I can make it a `not_if` though :)

Comment (andytson) [inline recipes/etc-local.rb]: no worries, I wasn't aware of that being the default. Can you rebase -i to clean up some of the adding and removing?

I see you've added the guard, which I'm happy to be removed based on what you confirmed

Comment (andytson) [inline recipes/etc-local.rb]: i.e delete d5c3bc7d3e5b98a92ab04871514e2805b0cc0092

Comment (kierenevans) [inline recipes/etc-local.rb]: Done :)
We can keep the guard, it makes sure!

Comment (andytson) [inline recipes/etc-local.rb]: k

**Comment by andytson** [comment]
The cron suite appears to fail with:

```
Recipe: magento-ng::cron
         * cron_d[magento-site1] action create (up to date)

           * Parent directory /etc/cron.d does not exist.
           ================================================================================
           Error executing action `create` on resource 'template[/etc/cron.d/magento-site1]'
           ================================================================================

           Chef::Exceptions::EnclosingDirectoryDoesNotExist
           ------------------------------------------------
           Parent directory /etc/cron.d does not exist.
```

**Comment by kierenevans** [comment]
I guess `cron::default` ( https://github.com/opscode-cookbooks/cron/blob/v1.4.2/recipes/default.rb ) needs to be in the runlist or included by the `magento-ng::cron` recipe.

I'm not sure which is the best approach but I've been favouring an `include_recipe` call from the recipe that wraps around another cookbook, so there's no surprises when someone uses the recipe expecting it to do everything.

I think I must have only ran the kitchen suite with centos-6.6 :|

**Comment by andytson** [comment]
I think in this case, it being in the recipe is acceptable tho strongly-coupled

**Comment by kierenevans** [comment]
I've added it into the recipe now so the directory should be created by the package installation.

**Pull request merged**